scrapy/utils/url.py

```diff
@@ -133,3 +133,15 @@ def strip_url(url, strip_credentials=True, strip_default_port=True, origin_only=
         '' if origin_only else parsed_url.query,
         '' if strip_fragment else parsed_url.fragment
     ))
+
+
+def is_uri(text, protocols={'http', 'https', 'file'}):
```

Comment:
:lipstick: It may be better to make `protocols` keyword only:

```suggestion
def is_uri(text, *, protocols={'http', 'https', 'file'}):
```

Also, maybe it should be `schemes` instead of `protocols`.